web-server/src/grading-queue/create-immediate.ts

```diff
@@ -0,0 +1,26 @@
+import { addReservation, generateGradingJobFromConfig } from "../utils/helpers";
+import { redisSet } from "../utils/redis";
+import { GradingJobConfig } from "./types";
+
+const createImmediateJob = async (
+  gradingJobConfig: GradingJobConfig,
+): Promise<Error | null> => {
```

Comment:
I think it would be better to `throw` the errors in these situation rather `return` them.

Reply:
Will address in a future PR.